api-put-object-common.go

```diff
@@ -58,6 +58,10 @@ func optimalPartInfo(objectSize int64) (totalPartsCount int, partSize int64, las
 	if objectSize == -1 {
 		objectSize = maxMultipartPutObjectSize
 	}
+	// object size is '0' set default parts.
+	if objectSize == 0 {
+		return 1, 0, 0, nil
+	}
```

Comment:
`putObjectNoChecksum` is called when `size` is zero. So, this check is not needed.